File: 118/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans(numRows, vector<int>());
        ans[0].resize(1);
        ans[0][0] = 1;
        for(int i=1; i<numRows; i++) {
            ans[i].resize(i+1);
            ans[i][0] = 1; //左邊的1
            for(int j=1; j<i; j++){
                ans[i][j] = ans[i-1][j-1] + ans[i-1][j];
            }
            ans[i][i] = 1; //右邊的1
        }
        return ans;
    }
};

File: 118/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []
        for i in range(numRows):
            row = [1]*(i+1) # 先放一堆1,這樣頭尾的1就準備好了
            ans.append(row)
            for j in range(1,i): # 把中間的數字補起來
                ans[i][j] = ans[i-1][j-1] + ans[i-1][j]
        return ans
